$OpenBSD: patch-SSL_pm,v 1.1.6.2 2010/12/09 17:51:16 jasper Exp $

Security fix for SA42508,
Perl IO::Socket::SSL "verify_mode" Security Bypass Security Issue

Backported from IO-Socket-SSL 1.35 and 1.37.

--- SSL.pm.orig	Tue Dec  7 19:31:43 2010
+++ SSL.pm	Tue Dec  7 19:31:39 2010
@@ -1368,14 +1369,10 @@ sub new {
 
 	my $verify_mode = $arg_hash->{SSL_verify_mode};
 	if ( $verify_mode != Net::SSLeay::VERIFY_NONE() and 
+	    ( defined $arg_hash->{SSL_ca_file} || defined $arg_hash->{SSL_ca_path}) and
 		! Net::SSLeay::CTX_load_verify_locations( 
 			$ctx, $arg_hash->{SSL_ca_file} || '',$arg_hash->{SSL_ca_path} || '') ) {
-		if ( ! $arg_hash->{SSL_ca_file} && ! $arg_hash->{SSL_ca_path} ) {
-			carp("No certificate verification because neither SSL_ca_file nor SSL_ca_path known");
-			$verify_mode = Net::SSLeay::VERIFY_NONE();
-		} else {
-			return IO::Socket::SSL->error("Invalid certificate authority locations");
-		}
+		return IO::Socket::SSL->error("Invalid certificate authority locations");
 	}
 
 	if ($arg_hash->{'SSL_check_crl'}) {
